src/components/OTP.vue

```diff
@@ -0,0 +1,308 @@
+  methods: {
+    /** Determines how the input box should look.
+     * - If an input error needs to be displayed, the box has a a red border.
+     * - Otherwise, it has an opacity of 30.
+     * @param {Number} index - index of the input box
+     */
+    selectInputBoxClasses() {
+      return [
+        {
+          "border-red-600 focus:border-red-600": this.invalidInputMessage,
+          "pointer-events-none opacity-30": !this.invalidInputMessage,
+        },
+      ];
+    },
+
+    /** Calls the mapping function to validate the typed character
+     * @param {Object} event - event triggered when a character is typed
+     */
+    isValidEntry(event) {
+      if (
+        validationTypeToFunctionMap[this.programData["input"]["basicValidationType"]](
+          event
+        )
+      ) {
+        return true;
+      } else event.preventDefault();
+    },
+
+    /** Resets the invalid input message */
+    resetInvalidInputMessage() {
+      this.invalidInputMessage = "";
+    },
+
+    /** This function is called whenever something is entered in the input box.
+     * It checks if the required number of characters are being typed.
+     * @param {Object} event - the event which triggered this function
+     * @param {Number} index - the index of the input field
+     */
+    updateUserId(event) {
+      if (event.target.value.length == 0) {
+        this.invalidInputMessage = "";
+      } else if (event.target.value.length < this.programData["input"]["maxLengthOfId"]) {
+        this.invalidInputMessage = this.programData["text"]["default"]["invalid"][
+          "input"
+        ];
+      } else {
+        this.resetInvalidInputMessage();
+      }
+      if (event.target.value.length > this.programData["input"]["maxLengthOfId"]) {
+        event.target.value = event.target.value.slice(
+          0,
+          this.programData["input"]["maxLengthOfId"]
+        );
+        this.userId = event.target.value.toString();
+      }
+    },
+
+    /** Function that calls the API to send an OTP
+     * If the status is 200, the OTP has been sent.
+     * Otherwise, retry
+     */
+    async sendOTP() {
+      const response = await OTPAuth.sendOTP(parseInt(this.userId));
+      if (response.status == 200) {
+        this.displayOTPMessage = mapSendStatusCodeToMessage(response.status.toString());
+        this.isOTPSent = true;
+      } else {
+        this.displayOTPMessage = mapSendStatusCodeToMessage(response.status.toString());
+        this.isOTPSent = false;
+      }
+    },
+
+    /** Function that verifies the OTP */
+    async verifyOTP() {
+      const response = await OTPAuth.verifyOTP(
+        parseInt(this.userId),
+        parseInt(this.OTPCode)
+      );
+      this.displayOTPMessage = mapVerifyStatusCodeToMessage[response.status.toString()];
+      if (response.status == 200) {
+        this.authenticate();
+      }
+    },
+
+    /** This method authenticates the phone number.
+     */
+    async authenticatePhoneNumber() {
```

Comment:
I got confused between `authenticate()` method and `authenticatePhoneNumber` method
Maybe rename `authenticate` to `authenticateAndRedirect`?